client/src/App.js

```diff
@@ -17,30 +17,70 @@ import Accounts from './screens/Accounts';
 import Welcome from './screens/Welcome';
 import { BeatLoader } from 'react-spinners';
 import { red } from '@material-ui/core/colors';
+import axios from 'axios';
+// import Cookies from 'js-cookies';
+
+const PrivateComponent = ({ component: Component, ...props }) => {
+	const isLoggedIn = localStorage.getItem('isLoggedIn');
 
-function App() {
 	return (
-		<Fragment>
-			<Router>
-				<Switch>
-					<Route path="/" exact component={Home} />
-					<Route path="/login" exact component={Login} />
-					<Route path="/user/:userId" exact component={Profile} />
-					<Route path="/contact-us" exact component={ContactUs} />
-					<Route path="/register" exact component={SignUp} />
-					<Route path="/welcome" exact component={Welcome} />
-					<Route path="/user/:userId/accounts" exact component={Accounts} />
-					<Route
-						path="/user/:userId/personal-info"
-						exact
-						component={PersonalInfo}
+		<Route
+			exact={true}
+			{...props}
+			render={(innerProps) => {
+				return isLoggedIn ? (
+					<Component {...innerProps} />
+				) : (
+					<Redirect
+						to={{ pathname: '/login', state: { from: props.location } }}
 					/>
-					<Route component={NotFoundPage} />
-					<Redirect to="/404" />
-				</Switch>
-			</Router>
-		</Fragment>
+				);
+			}}
+		/>
 	);
+};
+
+class App extends React.Component {
+	componentDidMount() {
+		const isLoggedIn = localStorage.getItem('isLoggedIn');
+		if (isLoggedIn) {
+			const token = localStorage.getItem('token');
+			axios
+				.post('/check-authentication', { token })
+				.then((res) => {
+					console.log('auth', res);
+				})
+				.catch((err) => console.log(err));
+		}
```

Comment:
Bonus points: use earlier return to avoid the extra nesting depth and improve "squint test" score